File: downloads/calculator/calculator.js
// JavaScript console calculator
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

console.log("Simple JS Calculator");
function ask(){
  readline.question("Enter expression (or 'exit' to quit): ", input => {
    if(input.toLowerCase() === 'exit'){
      readline.close();
    } else {
      try { console.log("Result:", eval(input)); } 
      catch { console.log("Invalid input."); }
      ask();
    }
  });
}

ask();

File: downloads/calculator/calculator.cpp
#include <iostream>
#include <string>
#include <sstream>
using namespace std;

int main() {
    cout << "Simple C++ Calculator\n";
    string input;
    while (true) {
        cout << "Enter expression (or 'exit' to quit): ";
        getline(cin, input);
        if(input == "exit") break;
        try {
            stringstream ss(input);
            double a, b;
            char op;
            ss >> a >> op >> b;
            double result = 0;
            switch(op){
                case '+': result = a + b; break;
                case '-': result = a - b; break;
                case '*': result = a * b; break;
                case '/': result = a / b; break;
            }
            cout << "Result: " << result << endl;
        } catch(...) {
            cout << "Invalid input.\n";
        }
    }
}
